src/pages/components/PageDetailsPage/PageDetailsPage.tsx

```diff
@@ -142,148 +142,156 @@ const PageDetailsPage: React.FC<PageDetailsPageProps> = ({
-                      description: "page",
-                    },
-                    {
-                      date: localizeDate(data.publicationDate),
-                    },
-                  ),
-                  visibleLabel: intl.formatMessage({
-                    id: "X26jCC",
-                    defaultMessage: "Visible",
-                    description: "page label",
-                  }),
-                }}
-                onChange={change}
-              />
-              <CardSpacer />
-              <PageOrganizeContent
-                data={data}
-                errors={errors}
-                disabled={loading}
-                pageTypes={pageTypes}
-                pageType={data.pageType}
-                pageTypeInputDisplayValue={data.pageType?.name || ""}
-                onPageTypeChange={handlers.selectPageType}
-                fetchPageTypes={fetchPageTypes}
-                fetchMorePageTypes={fetchMorePageTypes}
-                canChangeType={!page?.pageType}
-              />
-            </div>
-          </Grid>
-          <Savebar
-            disabled={isSaveDisabled}
-            state={saveButtonBarState}
-            onCancel={() => navigate(pageListUrl())}
-            onDelete={page === null ? undefined : onRemove}
-            onSubmit={submit}
-          />
-          {canOpenAssignReferencesAttributeDialog && (
-            <AssignAttributeValueDialog
-              entityType={getReferenceAttributeEntityTypeFromAttribute(
-                assignReferencesAttributeId,
-                data.attributes,
-              )}
-              confirmButtonState={"default"}
-              products={referenceProducts}
-              pages={referencePages}
-              hasMore={handlers.fetchMoreReferences?.hasMore}
-              open={canOpenAssignReferencesAttributeDialog}
-              onFetch={handlers.fetchReferences}
-              onFetchMore={handlers.fetchMoreReferences?.onFetchMore}
-              loading={handlers.fetchMoreReferences?.loading}
-              onClose={onCloseDialog}
-              onSubmit={attributeValues =>
-                handleAssignReferenceAttribute(attributeValues, data, handlers)
-              }
+                <CardSpacer />
+                <SeoForm
+                  errors={errors}
+                  allowEmptySlug={!pageExists}
+                  description={data.seoDescription}
+                  disabled={loading}
+                  descriptionPlaceholder={""} // TODO: Cast description to string and trim it
+                  onChange={change}
+                  slug={data.slug}
+                  slugPlaceholder={data.title}
+                  title={data.seoTitle}
+                  titlePlaceholder={data.title}
+                  helperText={intl.formatMessage({
+                    id: "jZbT0O",
+                    defaultMessage:
+                      "Add search engine title and description to make this page easier to find",
```

Comment:
same here, I know that perhaps you are not the author, but when you touch this, we can leave it improved : )

Reply:
and in each place like this in your PR

Reply:
Definitely, you are right, I've updated it also.